### PR TITLE
New integration tests development: API option to rotate logs depending on file size

### PR DESCRIPTION
**Pull request by RamosFe**
|Related issue|
|---|
|https://github.com/wazuh/wazuh-qa/issues/2949|

## Description

Adds new integration tests for the API options related to log rotation based on file size.

### CODE REVIEW

**Review by EduLeon12 — DISMISSED** [review]
Please clean the `__innit__` file.  Move the test path configs to a utils file and import it from there. 

And the dictionary mapping should either be in the same utils file mentioned before or if it is intended to be used for future test cases, which I think it can be, move it to the [`qa-integration-framework` api constants](https://github.com/wazuh/qa-integration-framework/blob/main/src/wazuh_testing/constants/api.py) .

**Comment by RamosFe** [comment]
Moved the mapping dictionary to his own file `utils.py`.

**Review by GGP1 — CHANGES_REQUESTED** [review]
There are tests for the `size` option but not for the `enabled` option. We should implement these two:

1. Check that when `enabled` is `False`, the API logs rotate at midnight after one entry.
2. Check if it's `True`, the API logs will not rotate at midnight.

File: tests/integration/test_api/test_config/test_logs_rotation/data/test_cases/cases_sizes_options.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+- name: VALID_UPPERCASE_VALUE_FOR_SIZE_OPTION_IN_KB
+  description: Upload a valid API configuration with the log size value in uppercase in KB
+  configuration_parameters:
+    ENABLED: yes
+    SIZE: 2000K
+  metadata:
+    expected_error_code: 0
+    expected_message: "Listening on"
+
```
[the remark this answers is not in the document]
Comment (RamosFe) [reply tests/integration/test_api/test_config/test_logs_rotation/data/test_cases/cases_sizes_options.yaml:45]: Fixed in 34258c20c21ec76d8c6fc97902826e0f0cefa277.

The missing tests are related to the topic described in https://github.com/wazuh/wazuh-qa/issues/2949#issuecomment-1932685555.

**Review by GGP1 — APPROVED** [review]
LGTM

File: tests/integration/test_api/test_config/test_logs_rotation/test_logs_rotation_by_size.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+    file.delete_path_recursively(api_logs_folder)
+
+
+@pytest.mark.tier(level=0)
+@pytest.mark.parametrize('test_configuration,test_metadata', zip(test_configuration, test_metadata), ids=test_cases_id)
+def test_logs_rotate_to_expected_path(test_configuration, test_metadata, add_configuration, truncate_monitored_files,
+                                      delete_api_logs_folder_contents, daemons_handler, wait_for_api_start):
+    """
+    description: Check if the log rotates based by the file size
```
Comment (Selutario) [inline tests/integration/test_api/test_config/test_logs_rotation/test_logs_rotation_by_size.py:100]:
```suggestion
    description: Check if the log rotates based on the file size.
```

File: tests/integration/test_api/test_config/test_logs_rotation/test_logs_size_option.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+test_configuration, test_metadata, test_cases_id = get_test_cases_data(test_cases_path)
+test_configuration = load_configuration_template(test_configuration_path, test_configuration, test_metadata)
+
+
+@pytest.mark.tier(level=0)
+@pytest.mark.parametrize('test_configuration,test_metadata', zip(test_configuration, test_metadata), ids=test_cases_id)
+def test_logs_rotation_size_option_values(test_configuration, test_metadata, add_configuration,
+                                          truncate_monitored_files, capfd):
+    """
+    description: Check if the API works as expected with different values of log rotation size
```
Comment (Selutario) [inline tests/integration/test_api/test_config/test_logs_rotation/test_logs_size_option.py:85]:
```suggestion
    description: Check if the API works as expected with different values of log rotation size.
```

**Review by Selutario — CHANGES_REQUESTED** [review]
It looks good. Just some minor changes in the docstrings description.

File: tests/integration/test_api/test_config/test_logs_rotation/test_logs_size_option.py (hunk 3 above)
Comment (RamosFe) [reply tests/integration/test_api/test_config/test_logs_rotation/test_logs_size_option.py]: Fixed typo in commit [bce3422].(https://github.com/wazuh/wazuh/pull/21776/commits/bce3422023e219c971baa91896a70c2a13678d31)

File: tests/integration/test_api/test_config/test_logs_rotation/test_logs_rotation_by_size.py (hunk 2 above)
Comment (RamosFe) [reply tests/integration/test_api/test_config/test_logs_rotation/test_logs_rotation_by_size.py]: Fixed typo in commit bce3422.

**Comment by Selutario** [comment]
These changes have nothing to do with the failed checks. The new tests passed in Github actions:
- https://github.com/wazuh/wazuh/actions/runs/8543871929/job/23408629394?pr=21776

**Pull request merged**